web/src/features/time/TimeController.tsx

```diff
@@ -3,28 +3,32 @@ import TimeAverageToggle from 'components/TimeAverageToggle';
 import TimeSlider from 'components/TimeSlider';
 import { useAtom, useAtomValue } from 'jotai';
 import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom';
 import { twMerge } from 'tailwind-merge';
 import trackEvent from 'utils/analytics';
 import { TimeAverages, TrackEvent } from 'utils/constants';
-import { getZoneTimezone, useGetZoneFromPath } from 'utils/helpers';
+import { getZoneTimezone } from 'utils/helpers';
 import {
   isHourlyAtom,
   selectedDatetimeIndexAtom,
-  timeAverageAtom,
+  useTimeAverageSync,
 } from 'utils/state/atoms';
 import { useIsBiggerThanMobile } from 'utils/styling';
 
 import TimeAxis from './TimeAxis';
 import TimeHeader from './TimeHeader';
 
 export default function TimeController({ className }: { className?: string }) {
-  const [timeAverage, setTimeAverage] = useAtom(timeAverageAtom);
   const isHourly = useAtomValue(isHourlyAtom);
   const [selectedDatetime, setSelectedDatetime] = useAtom(selectedDatetimeIndexAtom);
   const [numberOfEntries, setNumberOfEntries] = useState(0);
   const { data, isLoading: dataLoading } = useGetState();
   const isBiggerThanMobile = useIsBiggerThanMobile();
-  const zoneId = useGetZoneFromPath();
+  const { zoneId } = useParams<{
+    zoneId: string;
```

Comment:
Consider using the [RouteParams type](https://github.com/electricitymaps/electricitymaps-contrib/pull/7361/files#diff-2e5aec0d80c33c218d97cb48edb61724a5e25bfc33478de5907916a90a92accfR42) here